File: js/task1/S1_IF_02.js
// S1_IF_02 — Score to grade

function grade(score) {
  if (score < 0 || score > 100) {
    return null;
  }

  if (score >= 90) {
    return "A";
  } else if (score >= 80) {
    return "B";
  } else if (score >= 70) {
    return "C";
  } else if (score >= 60) {
    return "D";
  } else {
    return "F";
  }
}

// Tests
console.log(grade(95));   // A
console.log(grade(85));   // B
console.log(grade(75));   // C
console.log(grade(65));   // D
console.log(grade(45));   // F
console.log(grade(-5));   // null
console.log(grade(120));  // null

File: python/task1b/S1_IF_02.py
# S1_IF_02 — Score to grade

def grade(score):
    try:
        s = float(score)
    except (TypeError, ValueError):
        return None

    if s < 0 or s > 100:
        return None

    if s >= 90:
        return "A"
    elif s >= 80:
        return "B"
    elif s >= 70:
        return "C"
    elif s >= 60:
        return "D"
    else:
        return "F"


def main():
    tests = [59, 60, 69, 70, 89, 90, 100, 101]

    for t in tests:
        print(f"score={t:>3} -> grade={grade(t)}")


if __name__ == "__main__":
    main()
